packages/prettier-plugin-java/src/printers/blocks-and-statements.js

```diff
@@ -49,18 +49,23 @@ class BlocksAndStatementPrettierVisitor {
 
   localVariableDeclarationStatement(ctx) {
     const localVariableDeclaration = this.visit(ctx.localVariableDeclaration);
-    return rejectAndConcat([localVariableDeclaration, ctx.Semicolon[0]]);
+    return group(rejectAndConcat([localVariableDeclaration, ctx.Semicolon[0]]));
   }
 
   localVariableDeclaration(ctx) {
     const variableModifiers = this.mapVisit(ctx.variableModifier);
     const localVariableType = this.visit(ctx.localVariableType);
     const variableDeclaratorList = this.visit(ctx.variableDeclaratorList);
-
     return rejectAndJoin(" ", [
       rejectAndJoin(" ", variableModifiers),
-      localVariableType,
-      variableDeclaratorList
+      group(
+        concat([
+          variableModifiers.length > 0 ? softline : [],
```

Comment:
As rejectAndJoin will get rid of the variableModifiers if there are none, you can just use a line at line 59 instead of a " " separator, and group at an upper level.

Reply:
Currently, using annotation in a field declaration and in a local variable declaration have different bevhaviour:
As a field:
```Java
@RandomBreakingAnnotation(
  one = "One",
  two = "Two",
  three = "Three",
  four = "Four",
  five = "Five"
)
Integer[][] annotedArray = (Integer[][]) new Object[4][2];
```

as a local variable:
```Java
@RandomBreakingAnnotation(
  one = "One",
  two = "Two",
  three = "Three",
  four = "Four",
  five = "Five"
) Integer[][] annotedArray = (Integer[][]) new Object[4][2];
```

By changing the " " into a line and removing `variableModifiers.length > 0 ? softline : []`, we get thet first behaviour all the time. I think it is better to have an uniformized formatting but I think the second behaviour is prettier but that's my own opinion. "Fixing" the field declaration to make it look like the local variable declaration one will need another PR. WDYT

Reply:
I personally prefer the first one 😄

Reply:
Alright, I've gone along your review !